### PR TITLE
Update circleci config for a limited build on forks

### PR DESCRIPTION
**Pull request by avidan-H**
## Status
In Progress

## Related Issues
https://github.com/demisto/etc/issues/18464
https://github.com/demisto/etc/issues/17791

## Description
Run a limited build for forked PRs.

## Required version of Demisto
any

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Dependencies
Mention the dependencies of the entity you changed as given from the precommit hooks in checkboxes, and tick after tested them.
- [ ] Make sure the build process for unforked PRs is unchanged

### CODE REVIEW

**Comment by avidan-H** [comment]
Will need to update the circleci settings for the `content` repo to ensure that `Build forked pull requests` option is _on_ and that `Pass secrets to builds from forked pull requests` is _off_
image

File: .circleci/config.yml [hunk 1]
```diff
@@ -43,45 +59,70 @@ jobs:
...
             pytest ./Tests/scripts/test_configure_tests.py -v
+      - run:
+          name: Spell Checks
+          when: always
+          command: |
+            python ./Tests/scripts/circleci_spell_checker.py $CIRCLE_BRANCH
+      - run:
+          name: Common Server Documentation
+          when: always
+          command: ./Documentation/commonServerDocs.sh
```
Comment (ronykoz) [inline .circleci/config.yml]: Why do we want this in validation_and_testing?

**Comment by ronykoz** [comment]
@avidan-H did you check if we expose secret keys this way?

**Comment by glicht** [comment]
Can we keep the name of the job as `build` and not `validation_and_testing`?

image

File: .circleci/config.yml (hunk 1 above)
Comment (glicht) [reply .circleci/config.yml]: It is good to have it. To make sure the common server docs are generated properly. I've had cases that this failed as a result of changing common server.

**Comment by avidan-H** [comment]
> Can we keep the name of the job as `build` and not `validation_and_testing`?
> 
> image

Well it's actually two separate jobs so in practice it actually looks like so. 
image
It's no problem to change the name of the job. I think `build` will work (The only reason I'm not certain is that it looks like it is a reserved job name when one isn't using workflows)
image

but given that there are actually two jobs do you still want the first one to be called `build`?

**Comment by avidan-H** [comment]
> @avidan-H did you check if we expose secret keys this way?

@ronykoz I'm fairly certain we are fine in that regard if you look at the documentation [here](https://circleci.com/docs/2.0/oss/#security) and [here](https://circleci.com/docs/2.0/oss/#pass-secrets-to-builds-from-forked-pull-requests)

**Comment by ronykoz** [comment]
@avidan-H i think that GitHub requires a `build` to pass as well as you can see in the following image.
image

**Comment by Itay4** [comment]
@glicht @anara123 ok from you to merge ?

**Comment by glicht** [comment]
Not sure artifacts are being stored during the `build` stage. See: https://circleci.com/gh/demisto/content/29894#artifacts/containers/0

**Comment by Itay4** [comment]
@glicht @anara123 please review

**Pull request merged**